tests/test_endtoend.py

```diff
@@ -21,77 +21,89 @@ def setUp(self) -> None:
     def tearDown(self) -> None:
         self.__clean_states()
 
+    def assertEndToEnd(self, args: List[str], path:str, *, returncode:int=0):
+        process = subprocess.run(
+            [sys.executable, '-m', 'opustrainer', *args],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            encoding="utf-8")
+
+        self.assertEqual(process.returncode, returncode, msg=process.stderr)
+
+        # Useful to generate the data, hehe
+        if False and path.endswith('.expected.out'):
```

Comment:
Hmmm, so we keep the output generating code in here and just change it to `True` when the tests change?

Reply:
Yes. Or I can comment it out. Not sure why I did the if False thing, probably out of lazyness.